### PR TITLE
AppVeyor configuration

### PR DESCRIPTION
**Pull request by snoyberg**
This includes the changes from #176 (CC @ezyang), rebased against master, and adds in an AppVeyor configuration as requested in #176. I've checked that the AppVeyor configuration works (see https://ci.appveyor.com/project/snoyberg/hackage-security/build/1.0.1), but someone will need to enable AppVeyor for haskell/hackage-security to make this work.

### CODE REVIEW

**Comment by 23Skidoo** [comment]
I don't have admin access to this repo, so I can't enable AppVeyor. I think @hvr has.

**Comment by phadej** [comment]
I wonder if it makes sense to support `hackage-repo-tool` on windows at all?

**Comment by 23Skidoo** [comment]
The patch adding Windows support is not that big, so I think it's fine.

File: hackage-repo-tool/src/Hackage/Security/RepoTool/Util/IO.hs [hunk 1]
```diff
@@ -62,10 +69,16 @@ createSymbolicLink :: (FsRoot root, FsRoot root')
                    -> Path root' -- ^ Link location
                    -> IO ()
 createSymbolicLink linkTarget linkLoc = do
+#ifndef mingw32_HOST_OS
     createDirectoryIfMissing True (takeDirectory linkLoc)
     linkTarget' <- toAbsoluteFilePath linkTarget
     linkLoc'    <- toAbsoluteFilePath linkLoc
     Posix.createSymbolicLink linkTarget' linkLoc'
+#else
+    error $ "Cannot create symbolic links on Windows"
```
Comment (phadej) [inline hackage-repo-tool/src/Hackage/Security/RepoTool/Util/IO.hs]: @23Skidoo this makes  `symlink-cabal-local-repo` command fail. Not that it's useful on Windows, but I'd much happier disabling that codepath cleanly (i.e. disabling the command on windows), and not creating technical debt.

Comment (23Skidoo) [reply hackage-repo-tool/src/Hackage/Security/RepoTool/Util/IO.hs]: Fine, but this `#ifdef` will still be needed.

Comment (phadej) [reply hackage-repo-tool/src/Hackage/Security/RepoTool/Util/IO.hs]: by cleanly i mean: #ifdef the whole function, and see what breaks -> disable that with good error message.

**Comment by 23Skidoo** [comment]
BTW, `cabal-install` test suite uses `hackage-repo-tool`, so this could come in handy.

File: hackage-repo-tool/src/Hackage/Security/RepoTool/Util/IO.hs (hunk 1 above)
Comment (23Skidoo) [reply hackage-repo-tool/src/Hackage/Security/RepoTool/Util/IO.hs]: OK, I can look into it tomorrow.

**Comment by snoyberg** [comment]
@hvr I don't really mind if you want to go ahead and implement this yourself in a different way (23d8e9195620b6e4b5e270154a017e3146bc72f5), but a comment in this PR explaining your motivations for posterity, and so that as a contributor I know what the status of my PR is, would be greatly appreciated.

**Comment by hvr** [comment]
It would be great if the review concerns from @phadej could be resolved and this PR could be rebased atop the new cabal-based appveyor script I implemented in `master`.

**Comment by 23Skidoo** [comment]
Closing in favour of #210, which is a version of this PR with the points raised by @phadej and @hvr addressed.

**Pull request closed without merging**